tools/clang/include/clang/Basic/Attr.td

```diff
@@ -992,6 +992,18 @@ def HLSLNodeTrackRWInputSharing : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+def HLSLCBuffer : InheritableAttr {
+  let Spellings = []; // No spellings!
+  let Subjects = SubjectList<[CXXRecord]>;
+  let Documentation = [Undocumented];
+}
+
+def HLSLTessPatch : InheritableAttr {
```

Comment:
What about the geometry shader stream out objects (PointStream, LineStream, TriangleStream)?  These should also prevent long vectors in their template type argument, since that defines the signature outputs for the geometry shader.

Reply:
Yes, a separate attribute for GS stream seems best.